tests/accelerators/test_dp.py

```diff
@@ -127,17 +128,56 @@ def test_dp_test(tmpdir):
 
 @RunIf(min_gpus=2)
 def test_dp_training_step_dict(tmpdir):
-    """
-    This test verify dp properly reduce dictionaries
-    """
+    """ This test verifies that dp properly reduces dictionaries """
 
-    model = BoringModel()
+    class TestModel(BoringModel):
+
+        def train_dataloader(self):
+            return DataLoader(RandomDataset(32, 64), batch_size=2)
+
+        def val_dataloader(self):
+            return DataLoader(RandomDataset(32, 64), batch_size=2)
+
+        def test_dataloader(self):
+            return DataLoader(RandomDataset(32, 64), batch_size=2)
+
+        def add_outputs(self, output, device):
+            output.update({
+                "reduce_int": torch.tensor(device.index, dtype=torch.int, device=device),
+                "reduce_float": torch.tensor(device.index, dtype=torch.float, device=device),
+            })
+
+        def training_step(self, batch, batch_idx):
+            output = super().training_step(batch, batch_idx)
+            self.add_outputs(output, batch.device)
```

Comment:
btw, using `batch.device` here because `self.device` seems to be broken for DP

Reply:
yes it is, since only a copy of the model is moved in DP. Maybe we can still set the device to the root device in the plugin/wrapper?

Reply:
yes, already looking into that. I think it should be possible.